### PR TITLE
Split banner out

### PR DESCRIPTION
**Pull request by chrisnojima**
This splits the banner out into its own connected component

### CODE REVIEW

File: shared/chat/conversation/banner/container.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  ...getBannerMessage(state),
+})
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  onClick: (username: string) => { isMobile ? dispatch(onUserClick(username, '')) : dispatch(getProfile(username, true, true)) },
+})
+
+export default compose(
+  connect(mapStateToProps, mapDispatchToProps),
+  branch((props) => !props, renderNothing),
```
Comment (chrisnojima) [inline shared/chat/conversation/banner/container.js]: Moving this decision logic up so its shared, dumb items are dumber now

File: shared/chat/conversation/container.js [hunk 2]
```diff
@@ -162,31 +157,9 @@ const mapDispatchToProps = (dispatch: Dispatch, {setRouteState, navigateUp}) =>
 })
 
 const mergeProps = (stateProps, dispatchProps) => {
-  let bannerMessage
```
Comment (chrisnojima) [inline shared/chat/conversation/container.js]: 🔥

File: shared/chat/dumb.js [hunk 3]
```diff
@@ -411,62 +411,65 @@ const conversationsList = {
   },
 }
 
-const conversationBanner = {
-  component: ConversationBanner,
+const brokenTrackerBanner = {
```
Comment (chrisnojima) [inline shared/chat/dumb.js]: we have individual dumb components we render now

File: shared/constants/chat.js [hunk 4]
```diff
@@ -626,7 +626,7 @@ const getSelectedInbox = (state: TypedState) => {
 const getTLF = createSelector(
   [getSelectedInbox, getSelectedConversation],
   (selectedInbox, selected) => {
-    if (isPendingConversationIDKey(selected)) {
+    if (selected && isPendingConversationIDKey(selected)) {
```
Comment (chrisnojima) [inline shared/constants/chat.js]: this is in a different pr but i needed it also, just waiting on CI in that other pr

**Comment by chrisnojima** [comment]
@keybase/react-hackers one more piece split out from the main container

**Comment by keybase-ci-visdiff** [comment]
The commits 23a600cd538f...c6fea4ab70b3 introduce visual changes on linux. <details><summary>:mag_right: 6 new, 6 removed</summary>

 * Added: **Chat_Banner_Invite-Invite**  
   Chat_Banner_Invite-Invite rendered
 * Added: **Chat_Banner_Info-Info**  
   Chat_Banner_Info-Info rendered
 * Added: **Chat_Banner_Error-Error**  
   Chat_Banner_Error-Error rendered
 * Added: **Chat_Banner_Broken-Broken_Tracker_3**  
   Chat_Banner_Broken-Broken_Tracker_3 rendered
 * Added: **Chat_Banner_Broken-Broken_Tracker_2**  
   Chat_Banner_Broken-Broken_Tracker_2 rendered
 * Added: **Chat_Banner_Broken-Broken_Tracker_1**  
   Chat_Banner_Broken-Broken_Tracker_1 rendered
 * Removed: **Chat_Banner-Invite** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1491340228773-23a600cd538f-c6fea4ab70b3-linux/Chat_Banner-Invite-23a600cd538f.png)
 * Removed: **Chat_Banner-Info** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1491340228773-23a600cd538f-c6fea4ab70b3-linux/Chat_Banner-Info-23a600cd538f.png)
 * Removed: **Chat_Banner-Error** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1491340228773-23a600cd538f-c6fea4ab70b3-linux/Chat_Banner-Error-23a600cd538f.png)
 * Removed: **Chat_Banner-Broken_Tracker_3** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1491340228773-23a600cd538f-c6fea4ab70b3-linux/Chat_Banner-Broken_Tracker_3-23a600cd538f.png)
 * Removed: **Chat_Banner-Broken_Tracker_2** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1491340228773-23a600cd538f-c6fea4ab70b3-linux/Chat_Banner-Broken_Tracker_2-23a600cd538f.png)
 * Removed: **Chat_Banner-Broken_Tracker_1** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1491340228773-23a600cd538f-c6fea4ab70b3-linux/Chat_Banner-Broken_Tracker_1-23a600cd538f.png)
</summary>

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chrisnojima** [comment]
Visdiff changes are correct

**Comment by chrisnojima** [comment]
fix merge conflict
- [ ] Move validated to list

**Comment by keybase-ci-visdiff** [comment]
The commits b8eeb132bb69...29bbed635a08 introduce visual changes on linux. <details><summary>:mag_right: 6 new, 6 removed</summary>

 * Added: **Chat_Banner_Invite-Invite**  
   Chat_Banner_Invite-Invite rendered
 * Added: **Chat_Banner_Info-Info**  
   Chat_Banner_Info-Info rendered
 * Added: **Chat_Banner_Error-Error**  
   Chat_Banner_Error-Error rendered
 * Added: **Chat_Banner_Broken-Broken_Tracker_3**  
   Chat_Banner_Broken-Broken_Tracker_3 rendered
 * Added: **Chat_Banner_Broken-Broken_Tracker_2**  
   Chat_Banner_Broken-Broken_Tracker_2 rendered
 * Added: **Chat_Banner_Broken-Broken_Tracker_1**  
   Chat_Banner_Broken-Broken_Tracker_1 rendered
 * Removed: **Chat_Banner-Invite** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1491515512743-b8eeb132bb69-29bbed635a08-linux/Chat_Banner-Invite-b8eeb132bb69.png)
 * Removed: **Chat_Banner-Info** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1491515512743-b8eeb132bb69-29bbed635a08-linux/Chat_Banner-Info-b8eeb132bb69.png)
 * Removed: **Chat_Banner-Error** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1491515512743-b8eeb132bb69-29bbed635a08-linux/Chat_Banner-Error-b8eeb132bb69.png)
 * Removed: **Chat_Banner-Broken_Tracker_3** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1491515512743-b8eeb132bb69-29bbed635a08-linux/Chat_Banner-Broken_Tracker_3-b8eeb132bb69.png)
 * Removed: **Chat_Banner-Broken_Tracker_2** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1491515512743-b8eeb132bb69-29bbed635a08-linux/Chat_Banner-Broken_Tracker_2-b8eeb132bb69.png)
 * Removed: **Chat_Banner-Broken_Tracker_1** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1491515512743-b8eeb132bb69-29bbed635a08-linux/Chat_Banner-Broken_Tracker_1-b8eeb132bb69.png)
</summary>

**Pull request merged**